waku/v2/protocol/store/waku_store_protocol_test.go

```diff
@@ -425,3 +425,86 @@ func TestWakuStoreStart(t *testing.T) {
 	defer s2.Stop()
 
 }
+
+func TestWakuStoreWithStaticSharding(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	host1, err := libp2p.New(libp2p.DefaultTransports, libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0"))
+	require.NoError(t, err)
+
+	s1 := NewWakuStore(MemoryDB(t), nil, timesource.NewDefaultClock(), prometheus.DefaultRegisterer, utils.Logger())
+	s1.SetHost(host1)
+
+	// Prepare pubsub topics for static sharding
+	pubSubTopics := protocol.ShardsToTopics(20, []int{1, 2, 3})
+
+	// Prepare test messages
+	now := *utils.GetUnixEpoch()
+	msg1 := tests.CreateWakuMessage("hello", proto.Int64(now))
+
+	nowPlusOne := proto.Int64(now + 1)
+	msg2 := tests.CreateWakuMessage("/test/2/my-app/sharded", nowPlusOne)
+
+	nowPlusTwo := proto.Int64(now + 2)
+	msg3 := tests.CreateWakuMessage("/test/2/my-app/sharded", nowPlusTwo)
+
+	// Subscribe to pubSubtopics and start store1 + host1 with them
+	sub := SimulateSubscription([]*protocol.Envelope{
+		protocol.NewEnvelope(msg1, *utils.GetUnixEpoch(), pubSubTopics[0]),
+		protocol.NewEnvelope(msg2, *utils.GetUnixEpoch(), pubSubTopics[1]),
+		protocol.NewEnvelope(msg3, *utils.GetUnixEpoch(), pubSubTopics[2]),
+	})
+	err = s1.Start(ctx, sub)
+	require.NoError(t, err)
+	defer s1.Stop()
+
+	host2, err := libp2p.New(libp2p.DefaultTransports, libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0"))
+	require.NoError(t, err)
+
+	host2.Peerstore().AddAddr(host1.ID(), tests.GetHostAddress(host1), peerstore.PermanentAddrTTL)
+	err = host2.Peerstore().AddProtocols(host1.ID(), StoreID_v20beta4)
+	require.NoError(t, err)
+
+	s2 := NewWakuStore(MemoryDB(t), nil, timesource.NewDefaultClock(), prometheus.DefaultRegisterer, utils.Logger())
+	s2.SetHost(host2)
+
+	// Subscribe to DefaultWakuTopic at store2 + host2
+	sub1 := relay.NewSubscription(protocol.NewContentFilter(relay.DefaultWakuTopic))
```

Comment:
Did not understand why this is required? DefaultpubsubTopic is not related to static sharding.

Reply:
Is there another way how can I get empty subscription ? I need it for s2.Start(ctx, sub1), but I wish s2 doesn't know anything upfront about pubsub topics of s1.

Reply:
Not sure what you mean by empty sub, but you can always ise a different pubsubTopic in same cluster that is not being used by s1.

Reply:
New topic added at https://github.com/waku-org/go-waku/pull/1060/commits/12a8923d6270d4e185a683607bb51fe1d3beb987